addOns/wappalyzer/src/main/java/org/zaproxy/zap/extension/wappalyzer/WappalyzerPassiveScanner.java

```diff
@@ -48,6 +48,36 @@ public class WappalyzerPassiveScanner implements PassiveScanner {
     private WappalyzerApplicationHolder applicationHolder;
     private Set<String> visitedSiteIdentifiers = Collections.synchronizedSet(new HashSet<>());
     private volatile boolean enabled = true;
+    private volatile boolean exhaustive = false;
+
+    /** Functional interface for looped processing of HttpMessages in different ways. */
+    @FunctionalInterface
+    private interface CustomMessageProcessor {
```

Comment:
IMO it would have been better to make them uniform units of work, would just need a single loop. Future changes anyway.

Reply:
I guess I could make it Object generically instead of HttpMessage/Source? Then just case it within the loop/call?

Reply:
I'd not go down that route.